runtime/parachains/src/inclusion.rs

```diff
@@ -413,15 +414,26 @@ impl<T: Trait> Module<T> {
 					Error::<T>::NotCollatorSigned,
 				);
 
-				check_cx.check_validation_outputs(
-					para_id,
-					&candidate.candidate.commitments.head_data,
-					&candidate.candidate.commitments.new_validation_code,
-					candidate.candidate.commitments.processed_downward_messages,
-					&candidate.candidate.commitments.upward_messages,
-					T::BlockNumber::from(candidate.candidate.commitments.hrmp_watermark),
-					&candidate.candidate.commitments.horizontal_messages,
-				)?;
+				if let Err(err) = check_cx
+					.check_validation_outputs(
+						para_id,
+						&candidate.candidate.commitments.head_data,
+						&candidate.candidate.commitments.new_validation_code,
+						candidate.candidate.commitments.processed_downward_messages,
+						&candidate.candidate.commitments.upward_messages,
+						T::BlockNumber::from(candidate.candidate.commitments.hrmp_watermark),
+						&candidate.candidate.commitments.horizontal_messages,
+					)
+				{
+					log::debug!(
+						target: LOG_TARGET,
+						"Validation outputs checking during inclusion of a candidate {} for parachain `{}` failed: {:?}",
+						candidate_idx,
+						u32::from(para_id),
+						err,
+					);
+					Err(err.strip_into_dispatch_err::<T>())?;
```

Comment:
Is it desirable to continue checking candidate to determine all candidate validation failures, which can be collected into a single item, or is it fine to terminate early at the first candidate validation failure (current behavior)?

Reply:
Good idea for checking multiple candidates at once!

But I don't believe this is the current behavior though.

Reply:
I guess in this case, we would have to return the first or the last failure as a dispatch err.

Reply:
Yeah, I had in my head that we could return like a `Err(Vec<u32>)` with the indices, but if we have to strip that out as a dispatch error anyway, then there's not much point.